app/src/main/java/com/github/se/eduverse/model/folder/Folder.kt

```diff
@@ -0,0 +1,18 @@
+package com.github.se.eduverse.model.folder
+
+data class Folder(
+    val files: MutableList<MyFile>,
```

Comment:
I think that it's better to use a var of an immutable list instead of a mutable list because it ensures that the list can't be accidentally modified, making the code more predictable and safer. And it will be updated in a functional way which aligns with Kotlin's functional programming practices. Also I think the property should be a list of strings storing the folder's files ids as references to the files of the folder which will be stored in the db, and will be retrieved using these ids. The reason is that if we have a list of files as a property of Folder we will have two copies of the files in the db.

Reply:
The files argument is set as a mutable list so that it can be sorted, as it is most cost effective than creating a new list and storing it instead. Storing it in an immutable list would be better if the list wasn't meant to be changed.
I don't think the files should be stored fully on the database as it would necessite manipulating object on it, potentialy creating a lot of congestion and making the feature too much internet-dependant. Also we don't create copy of the files ; the MyFile data class only contains general informations like the name of the file or the last access time precisly in order to avoid access to the actual file everytime you need those informations.

Reply:
I agree with you about the type of the list it should be kept as MyFile. But I don't think that it would be costly to update the lists in a functional way but we can ask the coaches about that and keep it this way for now